### DISCUSSION TITLE
what is the normalization for json sources (like mongodb) to a relational database

### DISCUSSION THREAD

**Discussion by axellpadilla** [slingdata-io/sling-cli#341] [Q&A] [answered]
Hi and thanks ahead, searched for it, but couldn't find an answer to what sling does for column names and arrays on json objects, will it normalize and create extra tables if needed? will it clean names of columns?

**Comment by flarco** [comment, marked as answer]
Hi, by default it keeps the JSON as source, and does not flatten. If you set the `source_options.flatten` = `true`, then object keys will become their own columns (type inferred), and arrays will stay as it (as a JSON type column). Sling will not unnest.

**Reply by axellpadilla** [reply]
thank you, so also column names remain the same or is there an option to normalize?

**Reply by flarco** [reply]
Not sure what you mean by normalize, but there is the `column_casing` target option.
<img width="724" alt="image" src="https://github.com/user-attachments/assets/0fa9a5a9-b305-4954-adb6-fa3fc6ddce32">

**Reply by axellpadilla** [reply]
that's close, thanks, normalize is a step more than casing (remove spaces, special symbols, etc)

**Reply by flarco** [reply]
ah yes, it does: https://github.com/slingdata-io/sling-cli/blob/main/core/sling/task.go#L540
